### PR TITLE
[BE] My bills tab

### PR DESCRIPTION
**Pull request by TomaszKolek**
I also fixed voting - it was buggy a bit

### CODE REVIEW

File: project/api/models/account.py [hunk 1]
```diff
@@ -259,3 +261,28 @@ def is_full_account(self):
             return True
         else:
             return False
+
+    def get_voted_bills(self):
+        from .activity import Activity
```
Comment (TomaszKolek) [inline project/api/models/account.py]: its to avoid circular dependencies

File: project/api/write.py [hunk 2]
```diff
@@ -153,52 +154,31 @@ def rateCivi(request):
...
-            voted_civi.votes_neg = voted_civi.votes_neg + 1
-            vote_val = 'vote_neg'
-        elif rating == "neutral":
-            voted_civi.votes_neutral = voted_civi.votes_neutral + 1
-            vote_val = 'vote_neutral'
-        elif rating == "pos":
-            voted_civi.votes_pos = voted_civi.votes_pos + 1
-            vote_val = 'vote_pos'
-        elif rating == "vpos":
-            # c.votes_vpos = c.votes_vpos + 1
```
Comment (TomaszKolek) [inline project/api/write.py]: it was buggy commented I think

File: project/api/write.py [hunk 3]
```diff
@@ -153,52 +154,31 @@ def rateCivi(request):
...
+    }
 
-        activity_data = {
-            'account': account,
-            'thread': voted_civi.thread,
-            'civi': voted_civi,
-        }
+    activity_vote_key = 'votes_{}'.format(rating)
+    vote_val = 'vote_{}'.format(rating)
+    setattr(voted_civi, activity_vote_key, F(activity_vote_key) + 1)
```
Comment (TomaszKolek) [inline project/api/write.py]: It resolves a problem with voting race - the previous implementation could lose some data

File: project/api/models/account.py (hunk 1 above)
Comment (brylie) [reply project/api/models/account.py]: Add that as a comment to the code, so we don't forget.

File: project/api/models/account.py [hunk 4]
```diff
@@ -259,3 +261,28 @@ def is_full_account(self):
...
+
+        activities = Activity.objects.filter(account=self).prefetch_related(
+            'civi__linked_bills', 'civi__linked_civis__linked_bills'
+        )
+        supported_bills = []
+        opposed_bills = []
+
+        for activity in activities.iterator():
+            if activity.is_voted_negatively:
```
Comment (brylie) [inline project/api/models/account.py]: Can we rename this to `activity.is_negative_vote` / `activity.is_positive_vote`? It is a bit less verbose.

File: project/api/models/activity.py [hunk 5]
```diff
@@ -36,3 +36,11 @@ class Activity(models.Model):
 
     created = models.DateTimeField(auto_now_add=True, blank=True, null=True)
     last_modified = models.DateTimeField(auto_now=True, blank=True, null=True)
+
+    @property
+    def is_voted_positively(self):
```
Comment (brylie) [inline project/api/models/activity.py]: For an API, `is_positive_vote` and `is_negative_vote` sound a bit more 'natural'.

File: project/api/write.py (hunk 3 above)
Comment (brylie) [reply project/api/write.py]: Add that as a comment to the code.

**Comment by TomaszKolek** [comment]
@brylie review done

**Pull request merged**